src/bind-all.ts

```diff
@@ -35,254 +35,31 @@ function getBinding(original: Binding, sharedOptions?: boolean | AddEventListene
-    Binding<Target, Type8>,
-    Binding<Target, Type9>,
-  ],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<
-  Target extends EventTarget,
-  Type1 extends string,
-  Type2 extends string,
-  Type3 extends string,
-  Type4 extends string,
-  Type5 extends string,
-  Type6 extends string,
-  Type7 extends string,
-  Type8 extends string,
-  Type9 extends string,
-  Type10 extends string,
->(
-  target: Target,
-  bindings: [
-    Binding<Target, Type1>,
-    Binding<Target, Type2>,
-    Binding<Target, Type3>,
-    Binding<Target, Type4>,
-    Binding<Target, Type5>,
-    Binding<Target, Type6>,
-    Binding<Target, Type7>,
-    Binding<Target, Type8>,
-    Binding<Target, Type9>,
-    Binding<Target, Type10>,
-  ],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<
-  Target extends EventTarget,
-  Type1 extends string,
-  Type2 extends string,
-  Type3 extends string,
-  Type4 extends string,
-  Type5 extends string,
-  Type6 extends string,
-  Type7 extends string,
-  Type8 extends string,
-  Type9 extends string,
-  Type10 extends string,
-  Type11 extends string,
->(
-  target: Target,
-  bindings: [
-    Binding<Target, Type1>,
-    Binding<Target, Type2>,
-    Binding<Target, Type3>,
-    Binding<Target, Type4>,
-    Binding<Target, Type5>,
-    Binding<Target, Type6>,
-    Binding<Target, Type7>,
-    Binding<Target, Type8>,
-    Binding<Target, Type9>,
-    Binding<Target, Type10>,
-    Binding<Target, Type11>,
-  ],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<
-  Target extends EventTarget,
-  Type1 extends string,
-  Type2 extends string,
-  Type3 extends string,
-  Type4 extends string,
-  Type5 extends string,
-  Type6 extends string,
-  Type7 extends string,
-  Type8 extends string,
-  Type9 extends string,
-  Type10 extends string,
-  Type11 extends string,
-  Type12 extends string,
->(
-  target: Target,
-  bindings: [
-    Binding<Target, Type1>,
-    Binding<Target, Type2>,
-    Binding<Target, Type3>,
-    Binding<Target, Type4>,
-    Binding<Target, Type5>,
-    Binding<Target, Type6>,
-    Binding<Target, Type7>,
-    Binding<Target, Type8>,
-    Binding<Target, Type9>,
-    Binding<Target, Type10>,
-    Binding<Target, Type11>,
-    Binding<Target, Type12>,
-  ],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll(
-  target: EventTarget,
-  bindings: Binding[],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll(
-  target: EventTarget,
-  bindings: Binding[],
-  sharedOptions?: boolean | AddEventListenerOptions,
 ): UnbindFn {
-  const unbinds: UnbindFn[] = bindings.map((original: Binding) => {
-    const binding: Binding = getBinding(original, sharedOptions);
+  const unbinds: UnbindFn[] = bindings.map((original) => {
+    const binding: Binding = getBinding(original as never, sharedOptions);
```

Comment:
@Andarist can you please add a comment to the code explaining why `as never` is needed? I suspect that would be helpful information long term

Reply:
It's just a "fancy" `as any` - I can just change it to `as any` to avoid confusing people. 

`as never` works because `never` is assignable to just everything (but nothing, except `never` is assignable to it). Since the cast happens in the argument position - it truly doesn't matter which one we use here.

Reply:
You can go for whichever you think is best (and most maintainable)

Reply:
There is no real difference between both here. When types flow through layers you might end up with the casted type in the return type or something, so then `as never` might be a little bit safer than `as any` - because you won't accidentally start accessing properties on `any` type (as that would be `never` and you wouldn't be able to access any properties on it).

I decided to keep `as never` - as a slightly safer one (although not really at this particular call site at this specific point in time). If you'd prefer `as any` - I can change this any time